p4-14/v1.0.5/tex/p4.pt

```diff
@@ -705,7 +700,29 @@ References at run time to a header instance (or one of its fields) which is
 not valid results in a special ``undefined'' value.  The implications of this 
 depend on the context.
 \end{itemize}
-
+    
+To help programers write parsers for header stacks, P4 provides two
+keywords, \texttt{next} and \texttt{last}, that may be used to refer
+to header instances within a stack. These references are advanced
+automatically as the parser invokes \texttt{extract} in the parser.
+
+More precisely, the keyword \texttt{next} refers to the header
+instance at index \texttt{0} initially, and is advanced each time that
+\texttt{extract} is invoked on any instance within the stack. If
```

Comment:
Suppose someone does extract() on a header stack element with a constant index.  This sentence seems to imply that this should cause next to be advanced.  Is that the intended behavior?  I would have guessed that extract on header stack elements with constant indices would not advance next.